THBattle:
    #bugs: died person should drop all cards, should not respond to any input events

    ran: ui tune

    speaker, system_msg

    logic: n persons!

恶心丸不以弃牌方式进入弃牌堆，但恶心丸仍然生效...
魔导书出宴会，对方有西行妖，会卡主不动。。。 desync

显卡问题
双封魔阵

玩家离开提示
好人卡 时间提示
自动上传client_log.txt
大厅， 玩家状态
房间 打磨
五谷丰登 提示
装备区 card suit num
进度条 在tag上面 要改吗？

比如选人的时候 我都不知道选的什么人
强烈要求游戏开始前可以换队伍。

???
还有个 我不知道是BUG 还是什么
如果你2血 我伤害 2   你死了 9一下 1血
这是正常的。
如果 你1血 我伤害2   是不是打你-1血 你起码2个药吧
!! 但是1个药就活了

?5.蓝的2技能对罪袋狂欢和五谷丰登无效。弹幕战也无效。就是不会询问是否使用该技能再发动一次。

sinsack reject

reject prompt

等有空的时候我会加上让更新忽略文件的功能的

windows: update sometimes fail

?? 大妖精 开局 摸牌
?? 黑化芙兰弹幕战只扣一血
?? tewi 血量 死后摸牌。 其他人没有摸牌

ui_meta SOMETHING WRONG DollControl

?? 红三的技能点了直接跳过了诶=。=

Traceback (most recent call last):
  File "C:\Users\Proton\Desktop\thb\src\gamepack\thb\ui\inputs.py", line 191, in
 on_selection_change
    assert c == c1
AssertionError
INFO:Server:GAME_EXPECT: 'input_action_stage_usecard_182'

Traceback (most recent call last):
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/greenlet.py", line 390, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/proton/thbattle/src/server/core/client_endpoint.py", line 158, in _run
    self.write(['heartbeat', None])
  File "/home/proton/thbattle/src/network/endpoint.py", line 42, in write
    self.raw_write(self.encode(p))
  File "/home/proton/thbattle/src/network/endpoint.py", line 33, in raw_write
    self.sock.send(s)
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/socket.py", line 489, in send
    wait_write(sock.fileno(), timeout=timeout, event=self._write_event)
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/socket.py", line 188, in wait_write
    assert event.arg is None, 'This event is already used by another greenlet: %r' % (event.arg, )
AssertionError: This event is already used by another greenlet: (<_GameHallStatusUpdator at 0xb7133edcL>, timeout('timed out',))
<Client at 0xa010f2c> failed with AssertionError


?? 另外 被封魔阵封住的黑白应该也能偷的
bug：被罪袋炸死以后 身上如果有封魔状态不会清除



1.首轮自己行动之前不能发动（没有触发提示）
2.打个比方，比如A用罪呆狂欢杀死B，八云蓝再用机智重新攻击B触发游戏逻辑崩溃


天子给牌不能reveal all

tatara reveal all： need shuffle thing

id是？？？(虽然不知道为什么要id)建议：希望在游戏开始时有提示音。在点退出房间时有再次确定的选项。望采纳。
灵梦1血 剩3恶心丸 死了游戏崩溃

disable esc key


?
问下莱瓦顶是不是有BUG?
Proton  20:55:29
什么bug？
★突破天际の菜鸟★  20:55:44
貌似不出效果


Traceback (most recent call last):
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/greenlet.py", line 390, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/proton/thbattle/src/server/core/game_server.py", line 172, in _run
    self.game_start()
  File "/home/proton/thbattle/src/gamepack/thb/logic.py", line 197, in game_start
    self.process_action(PlayerTurn(p))
  File "/home/proton/thbattle/src/game/game_common.py", line 207, in process_action
    action = self.emit_event('action_apply', action)
  File "/home/proton/thbattle/src/game/game_common.py", line 169, in emit_event
    data = evt.handle(evt_type, data)
  File "/home/proton/thbattle/src/gamepack/thb/characters/ran.py", line 48, in handle
    Game.getgame().process_action(ProphetAction(tgt, tgt))
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/characters/ran.py", line 24, in apply_action
    if not rst: return act
NameError: global name 'act' is not defined
<THBattle at 0x89aa57c> failed with NameError


ERROR:THBattle_Actions:TryRevive buggy condition, __init__
ERROR:THBattle_Actions:TryRevive buggy condition, apply
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/greenlet.py", line 390, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/proton/thbattle/src/server/core/game_server.py", line 172, in _run
    self.game_start()
  File "/home/proton/thbattle/src/gamepack/thb/logic.py", line 197, in game_start
    self.process_action(PlayerTurn(p))
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 627, in apply_action
    g.process_action(DropCardStage(p))
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 343, in apply_action
    g.process_action(DropCards(target, cards=cards))
  File "/home/proton/thbattle/src/game/game_common.py", line 222, in process_action
    action = self.emit_event('action_after', action)
  File "/home/proton/thbattle/src/game/game_common.py", line 169, in emit_event
    data = evt.handle(evt_type, data)
  File "/home/proton/thbattle/src/gamepack/thb/cards/basic.py", line 152, in handle
    g.process_action(Damage(source=None, target=target))
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 267, in apply_action
    if not g.process_action(TryRevive(tgt)):
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 208, in apply_action
    traceback.print_stack()



  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 267, in apply_action
    if not g.process_action(TryRevive(tgt)):
  File "/home/proton/thbattle/src/game/game_common.py", line 210, in process_action
    rst = action.apply_action()
  File "/home/proton/thbattle/src/gamepack/thb/actions.py", line 208, in apply_action
    traceback.print_stack()
Traceback (most recent call last):
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/greenlet.py", line 390, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/proton/thbattle/src/server/core/client_endpoint.py", line 158, in _run
    self.write(['heartbeat', None])
  File "/home/proton/thbattle/src/network/endpoint.py", line 45, in write
    self.raw_write(self.encode(p))
  File "/home/proton/thbattle/src/network/endpoint.py", line 34, in raw_write
    self.sock.send(s)
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/socket.py", line 489, in send
    wait_write(sock.fileno(), timeout=timeout, event=self._write_event)
  File "/home/proton/env/local/lib/python2.7/site-packages/gevent/socket.py", line 188, in wait_write
    assert event.arg is None, 'This event is already used by another greenlet: %r' % (event.arg, )
AssertionError: This event is already used by another greenlet: (<_GameHallStatusUpdator at 0xb7049f2cL>, timeout('timed out',))
<Client at 0xb700bd9cL> failed with AssertionError


















曹　操
夏侯惇 船长
关　羽
吕　蒙
黄　盖

CommonUI:
ScrollBar
ARTWORKS!

Core:
user subsystem,
error reporting subsystem
    * server auto restart

身份场：
黑幕 -- 主
异变 -- 忠
解决者 -- 反
不明 -- 内

addtional:

catan -- ？
樱树 -- 春度
神社 -- 信仰
土地 -- 节操
？？ -- 威严
。。。。？

达芬奇密码 -- 笨蛋9的算术教室
牌背面 白莉莉和黑莉莉
